src/System.Linq/src/System/Linq/Range.cs

```diff
@@ -101,6 +101,28 @@ public List<int> ToList()
                 return list;
             }
 
+            public HashSet<int> ToHashSet(IEqualityComparer<int> comparer)
+            {
+                HashSet<int> hashSet;
+
+                // Pre-allocate only when default comparer is used
+                if (Utilities.AreEqualityComparersEqual(comparer, EqualityComparer<int>.Default))
```

Comment:
Nit: You could add a `using static System.Linq.Utilities` at the top and use the ternary operator here for concision

```cs
// Pre-allocate only when default comparer is used
HashSet<int> hashSet =
    AreEqualityComparersEqual(comparer, null) ?
    new HashSet<int>(_end - _start, comparer) :
    new HashSet<int>(comparer);
```